### PR TITLE
Added referrer-policy as an optional header

### PR DESCRIPTION
**Pull request by deg4uss3r**
## Description

Added `Referrer-Policy` header for enhancement #1  

## Motivation and Context

`Referrer-Policy` is a security header that helps prevent referrer leakages by controlling when the referrer is displayed. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "armor"
-version = "1.0.0"
+version = "1.0.1"
```
Comment (yoshuawuyts) [inline Cargo.toml]: Could we leave the version out of this PR? I prefer to version things as part of a release (which would also need to be _minor_, not _patch_ in this case).

Comment (deg4uss3r) [reply Cargo.toml]: Yup, my bad!

File: src/lib.rs [hunk 2]
```diff
@@ -144,3 +144,66 @@ pub fn dont_sniff_mimetype(headers: &mut HeaderMap) {
...
+/// armor::referrer_policy(&mut headers, Some(armor::ReferrerOptions::UnsafeUrl));
+/// armor::referrer_policy(&mut headers, Some(armor::ReferrerOptions::NoReferrer));
+/// let mut referrerValues: Vec<&str> = headers.get_all("Referrer-Policy").iter().map(|x| x.to_str().unwrap()).collect();
+/// assert_eq!(referrerValues.sort(), vec!("unsafe-url", "no-referrer").sort());
+/// ```
+#[inline]
+pub fn referrer_policy(headers: &mut HeaderMap, referrer: Option<ReferrerOptions>) {
+    let policy = match referrer {
+        None | Some(ReferrerOptions::Empty) => "",
+        Some(ReferrerOptions::NoReferrer) => "no-referrer",
```
Comment (yoshuawuyts) [inline src/lib.rs]: If we follow [helmet's lead](https://helmetjs.github.io/docs/referrer-policy/) here, this `no-referrer` should be the default.

File: src/lib.rs [hunk 3]
```diff
@@ -144,3 +144,66 @@ pub fn dont_sniff_mimetype(headers: &mut HeaderMap) {
...
+/// let mut headers = http::HeaderMap::new();
+/// armor::referrer_policy(&mut headers, Some(armor::ReferrerOptions::UnsafeUrl));
+/// armor::referrer_policy(&mut headers, Some(armor::ReferrerOptions::NoReferrer));
+/// let mut referrerValues: Vec<&str> = headers.get_all("Referrer-Policy").iter().map(|x| x.to_str().unwrap()).collect();
+/// assert_eq!(referrerValues.sort(), vec!("unsafe-url", "no-referrer").sort());
+/// ```
+#[inline]
+pub fn referrer_policy(headers: &mut HeaderMap, referrer: Option<ReferrerOptions>) {
+    let policy = match referrer {
+        None | Some(ReferrerOptions::Empty) => "",
```
Comment (yoshuawuyts) [inline src/lib.rs]: If people don't want to set the referrer at all, I think they could probably not use this function. Afaik a header with an empty value is the same as not sending a header at all.

File: src/lib.rs (hunk 2 above)
Comment (deg4uss3r) [reply src/lib.rs]: Added `no-referrer` as the default

File: src/lib.rs (hunk 3 above)
Comment (deg4uss3r) [reply src/lib.rs]: removed `ReferrerOptions::Empty` variant as well :)

**Review by yoshuawuyts — APPROVED** [review]
Looking great!

**Pull request merged**